File: 1160/solution.py
from collections import Counter


class Solution:
    def countCharacters(self, words: list[str], chars: str) -> int:
        chars_count = Counter(chars)
        res = 0
        for word in words:
            word_count = Counter(word)
            valid = True
            for key in word_count:
                if chars_count[key] < word_count[key]:
                    valid = False
                    break
            res += len(word) if valid else 0
        return res
        # time O(c + wc)
        # space O(1)

File: 1160/solution.go
package main

func largestGoodInteger(num string) string {
    res := ""
    for i := 0; i < len(num) - 2; i++ {
        if num[i] == num[i + 1] && num[i + 1] == num[i + 2] {
            if (len(res) == 0 || res[0] < num[i]) {
                res = num[i:i + 3]
            }
        }
    }
    return res
}